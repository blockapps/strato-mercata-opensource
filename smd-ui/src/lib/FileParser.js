export function getImportStatements(source) {
    const splittedSource = source.split("\n")
    // TODO: This does not guard against multiline strings or comments
    const newImports = splittedSource.filter((line) => /^\s*import /.test(line))
    const newFileList = []
    newImports.map((item) => {
        const splittedItem = item.split(' ')
        const filename = splittedItem.filter((word) => word.includes('.sol'))
        newFileList.push(filename.length > 0 && filename[0].replace(/['";]+/g, ''))
        return item
    })
    return newFileList
}

export function replaceImportStatementsWithSource(source, importsCode, tabData) {
    const splittedSource = source.split("\n")
    const filteredList = splittedSource.map((item) => {
        const code = importsCode.filter(
            (codeItem) =>
                item.includes(codeItem.title)
        )
        return code.length > 0 ? getFileAndReplaceWithImport(code[0].text, tabData) : item }
    )
    filteredList.join("\r\n")
    return filteredList
}

export function getFileAndReplaceWithImport(code, tabData) {
    const newFileList = getImportStatements(code)
    if (newFileList.length === 0) {
        return code
    }
    const items = tabData.filter((item) =>
        newFileList.includes(item.title)
    )
    const replacedCode = replaceImportStatementsWithSource(code, items, tabData)
    return replacedCode.join("\r\n")
}
