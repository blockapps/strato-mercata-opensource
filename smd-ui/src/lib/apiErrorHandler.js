export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
