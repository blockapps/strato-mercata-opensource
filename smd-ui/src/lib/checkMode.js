import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}