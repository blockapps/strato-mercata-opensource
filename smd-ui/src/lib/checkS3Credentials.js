import { env } from "../env";

export function isS3Available() {
  return env.EXT_STORAGE_ENABLED === 'true'
}
