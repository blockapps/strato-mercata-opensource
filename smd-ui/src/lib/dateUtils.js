import moment from 'moment';

export function parseDateFromString(timeAsString) {
  const utcDate = timeAsString.replace(' UTC', 'Z')
  const formattedDate = moment(utcDate).format('YYYY-MM-DD hh:mm:ss A')
  return formattedDate
}

export function parseDateFromTimestamp(timestamp) {
  return parseDateFromString(moment.unix(timestamp).utc().toString());
}
