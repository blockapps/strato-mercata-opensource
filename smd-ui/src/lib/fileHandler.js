export function downloadFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(content));
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}

export function downloadPDFFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', content);
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}