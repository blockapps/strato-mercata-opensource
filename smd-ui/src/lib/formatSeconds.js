import React from "react";
export function sec2Date(sec){
    if (sec === undefined || sec <= 0 ){
        return <div></div>;
    }
    let seconds = sec;
    const days = Math.floor(seconds / (3600*24));
    const daysString = days.toString().padStart(2, "0");
    seconds  -= days*3600*24;
    const hrs = Math.floor(seconds / 3600);
    const hoursString = hrs.toString().padStart(2, "0");
    seconds  -= hrs*3600;
    const mnts = Math.floor(seconds / 60);
    const minutesString = mnts.toString().padStart(2, "0");
    seconds  -= mnts*60;
    const secondsString = Math.floor(seconds).toString().padStart(2, "0");
    const ret = <div>
        Uptime {days > 0 ? <span>{daysString}d:</span> : null}
         <span>
             {hoursString}:
        </span>
        <span>
             {minutesString}:

        </span>
        <span>
             {secondsString}
        </span>
    </div>
    return ret;
}
