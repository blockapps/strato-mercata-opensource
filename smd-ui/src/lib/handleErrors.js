import { env } from "../env";

const HTTP_UNAUTHORIZED = 401;

export function handleErrors(response) {
  // Handled: UNAUTHORIZED and OAUTH_ENABLED
  if (env.OAUTH_ENABLED && response.status === HTTP_UNAUTHORIZED) {
    window.location.reload();
  }
  return response;
}