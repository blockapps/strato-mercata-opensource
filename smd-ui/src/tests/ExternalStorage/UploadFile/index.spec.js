import React from 'react';
import UplaodFile, { mapStateToProps } from '../../../components/ExternalStorage/UploadFile';
import { createStore, combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import * as checkMode from '../../../lib/checkMode';

describe('UploadFile: index', () => {
  let store

  beforeEach(() => {
    store = createStore(combineReducers({ form: formReducer }));
  });

  describe('render component (non Oauth mode)', () => {
    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
    })

    test('without values', () => {
      const props = {
        isOpen: false,
        uploadError: null,
        result: null,
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      ).find('Dialog');

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        isOpen: true,
        uploadError: null,
        result: {
          contractAddress: '23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c',
          uri: 'https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg',
          metadata: 'widescreen is one of the most important factor'
        },
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      ).find('Dialog');

      expect(wrapper.debug()).toMatchSnapshot();
    });

  });

  describe('render component (Oauth mode)', () => {
    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(true);
    })

    test('without values', () => {
      const props = {
        isOpen: false,
        uploadError: null,
        result: null,
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      ).find('Dialog');

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        isOpen: true,
        uploadError: null,
        result: {
          contractAddress: '23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c',
          uri: 'https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg',
          metadata: 'widescreen is one of the most important factor'
        },
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      ).find('Dialog');

      expect(wrapper.debug()).toMatchSnapshot();
    });

  });

  describe('Oauth Mode', () => {

    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(true);
    })

    test('componentWillReceiveProps', () => {
      const props = {
        isOpen: true,
        uploadError: null,
        result: {},
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      );

      wrapper.setProps({ uploadError: 'error' });

      expect(props.resetError).toHaveBeenCalledTimes(1);
      expect(props.resetError).toHaveBeenCalled();
    });

  });

  describe('Non Oauth Mode', () => {

    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
    })

    test('componentDidMount', () => {
      const props = {
        isOpen: true,
        uploadError: 'error',
        result: {},
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      );

      expect(props.fetchAccounts).toHaveBeenCalledTimes(1);
      expect(props.fetchAccounts).toHaveBeenCalled();
    });

    test('componentWillReceiveProps', () => {
      const props = {
        isOpen: true,
        uploadError: null,
        result: {},
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      );

      wrapper.setProps({ uploadError: null });

      expect(props.resetError).toHaveBeenCalledTimes(0);
      expect(props.resetError).not.toHaveBeenCalled();
    });

  });

  describe('mapStateToProps', () => {

    test('without values', () => {
      const state = {
        uploadFile: {
          isOpen: false,
          error: null,
          result: null
        }
      }

      expect(mapStateToProps(state)).toMatchSnapshot();
    });

    test('with values', () => {
      const state = {
        uploadFile: {
          isOpen: true,
          error: 'error',
          result: {
            contractAddress: '23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c',
            uri: 'https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg',
            metadata: 'widescreen is one of the most important factor'
          }
        }
      }

      expect(mapStateToProps(state)).toMatchSnapshot();
    });

  });

  describe('Dialog', () => {

    test('simulate onClose', () => {
      const props = {
        isOpen: true,
        uploadError: null,
        result: {
          contractAddress: '23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c',
          uri: 'https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg',
          metadata: 'widescreen is one of the most important factor'
        },
        closeUploadModal: jest.fn(),
        resetError: jest.fn(),
        fetchAccounts: jest.fn()
      }

      const wrapper = shallow(
        <UplaodFile.WrappedComponent {...props} />
      );

      let Dialog = wrapper.find('Dialog').dive().get(0);
      Dialog.props.onClose();

      expect(props.closeUploadModal).toHaveBeenCalledTimes(1);
      expect(props.closeUploadModal).toHaveBeenCalled();
    });

  });

});