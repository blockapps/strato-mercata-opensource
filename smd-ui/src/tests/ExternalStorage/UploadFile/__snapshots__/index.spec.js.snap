// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFile: index mapStateToProps with values 1`] = `
Object {
  "isOpen": true,
  "result": Object {
    "contractAddress": "23fe0e0ab7bd95fbcdff877660c595d24c6dcf5c",
    "metadata": "widescreen is one of the most important factor",
    "uri": "https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg",
  },
  "uploadError": "error",
}
`;

exports[`UploadFile: index mapStateToProps without values 1`] = `
Object {
  "isOpen": false,
  "result": null,
  "uploadError": null,
}
`;

exports[`UploadFile: index render component (Oauth mode) with values 1`] = `
"<Blueprint.Dialog isOpen={true} onClose={[Function: onClose]} iconName=\\"saved\\" title=\\"URI Upload Success\\" className=\\"pt-dark upload-dialog\\" canOutsideClickClose={true}>
  <UploadData result={{...}} closeModal={[Function: closeModal]} />
</Blueprint.Dialog>"
`;

exports[`UploadFile: index render component (Oauth mode) without values 1`] = `
"<Blueprint.Dialog isOpen={false} onClose={[Function: onClose]} iconName=\\"pt-icon-upload\\" title=\\"Upload & attest\\" className=\\"pt-dark upload-dialog\\" canOutsideClickClose={true}>
  <withRouter(Connect(ReduxForm)) />
</Blueprint.Dialog>"
`;

exports[`UploadFile: index render component (non Oauth mode) with values 1`] = `
"<Blueprint.Dialog isOpen={true} onClose={[Function: onClose]} iconName=\\"saved\\" title=\\"URI Upload Success\\" className=\\"pt-dark upload-dialog\\" canOutsideClickClose={true}>
  <UploadData result={{...}} closeModal={[Function: closeModal]} />
</Blueprint.Dialog>"
`;

exports[`UploadFile: index render component (non Oauth mode) without values 1`] = `
"<Blueprint.Dialog isOpen={false} onClose={[Function: onClose]} iconName=\\"pt-icon-upload\\" title=\\"Upload & attest\\" className=\\"pt-dark upload-dialog\\" canOutsideClickClose={true}>
  <withRouter(Connect(ReduxForm)) />
</Blueprint.Dialog>"
`;
